src/details/ArborX_DetailsConcepts.hpp

```diff
@@ -60,6 +61,45 @@ struct has_centroid<
 {
 };
 
+template <typename Primitives, typename = void>
+struct has_access_traits : std::false_type
+{
+};
+
+template <typename Primitives>
+struct has_access_traits<
+    Primitives,
+    std::void_t<
+        std::enable_if_t<Kokkos::is_memory_space<
+            typename Traits::Access<Primitives>::MemorySpace>::value>,
+        std::enable_if_t<std::is_same<typename Traits::Access<Primitives>::Tag,
+                                      PointTag>::value>,
+        std::enable_if_t<std::is_same<decltype(Traits::Access<Primitives>::get(
+                                          std::declval<Primitives const &>(),
+                                          std::declval<int>())),
+                                      Point>::value>,
+        decltype(Traits::Access<Primitives>::size(
+            std::declval<Primitives const &>()))>> : std::true_type
```

Comment:
It might be rather hard for a user to understand what the requirements are. What about something like https://godbolt.org/z/HHumU1?

Or are you planning on using SFINAE with `has_access_traits`?

Reply:
See newer version.